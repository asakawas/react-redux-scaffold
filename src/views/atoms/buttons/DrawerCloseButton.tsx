import React, { FunctionComponent } from 'react';
import IconButton, { IconButtonProps } from '@material-ui/core/IconButton';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import styled from 'styled-components';

const RightAligner = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
`;

/**
 * The type of props of DrawerCloseButton.
 */
export type DrawerCloseButtonProps = Readonly<{
  /** The event handler called when the close button is clicked */
  onDrawerCloseButtonClick: IconButtonProps['onClick'];
}>;

const DrawerCloseButton: FunctionComponent<DrawerCloseButtonProps> = ({
  onDrawerCloseButtonClick,
}) => (
  <RightAligner>
    <IconButton onClick={onDrawerCloseButtonClick} aria-label="drawer close button">
      <ChevronLeftIcon />
    </IconButton>
  </RightAligner>
);

export default DrawerCloseButton;
