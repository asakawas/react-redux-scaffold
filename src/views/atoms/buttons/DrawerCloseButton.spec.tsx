import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import DrawerCloseButton from './DrawerCloseButton';

describe('DrawerCloseButton', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<DrawerCloseButton onDrawerCloseButtonClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("calls the passed handler when it's clicked", () => {
    const handler = jest.fn();
    const renderResult = render(<DrawerCloseButton onDrawerCloseButtonClick={handler} />);

    const tree = renderResult.getByLabelText('drawer close button');
    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(1);

    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(2);
  });
});
