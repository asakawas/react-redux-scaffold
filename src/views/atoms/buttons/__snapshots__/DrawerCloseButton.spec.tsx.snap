// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerCloseButton is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
    class="c0"
  >
    <button
      aria-label="drawer close button"
      class="MuiButtonBase-root MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
